server/enterprise/metrics/metrics.go

```diff
@@ -1122,6 +1134,116 @@ func New(ps *platform.PlatformService, driver, dataSource string) *MetricsInterf
 	)
 	m.Registry.MustRegister(m.NotificationUnsupportedCounters)
 
+	m.ClientTimeToFirstByte = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: MetricsNamespace,
+			Subsystem: MetricsSubsystemClients,
+			Name:      "time_to_first_byte",
+			Help:      "Duration of when a browser starts to request a page from a server to when the server first starts to return data (milliseconds)",
+		},
+		[]string{"platform", "user_agent"},
+	)
+	m.Registry.MustRegister(m.ClientTimeToFirstByte)
+
+	m.ClientFirstContentfulPaint = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: MetricsNamespace,
+			Subsystem: MetricsSubsystemClients,
+			Name:      "first_contentful_paint",
+			Help:      "Duration of the load time of any content to be displayed on screen to a user (milliseconds)",
+		},
+		[]string{"platform", "user_agent"},
+	)
+	m.Registry.MustRegister(m.ClientFirstContentfulPaint)
+
+	m.ClientLargestContentfulPaint = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: MetricsNamespace,
+			Subsystem: MetricsSubsystemClients,
+			Name:      "largest_contentful_paint",
+			Help:      "Duration of the load time of largest content to be displayed on screen to a user (milliseconds)",
+		},
+		[]string{"platform", "user_agent"},
+	)
+	m.Registry.MustRegister(m.ClientLargestContentfulPaint)
+
+	m.ClientInteractionToNextPaint = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: MetricsNamespace,
+			Subsystem: MetricsSubsystemClients,
+			Name:      "interaction_to_next_paint",
+			Help:      "Duration of the longest click or keyboard interaction while app is running, ignoring outliers (milliseconds)",
+		},
+		[]string{"platform", "user_agent"},
+	)
+	m.Registry.MustRegister(m.ClientInteractionToNextPaint)
+
+	m.ClientCumulativeLayoutShift = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: MetricsNamespace,
+			Subsystem: MetricsSubsystemClients,
+			Name:      "cumulative_layout_shift",
+			Help:      "Duration of a page's content shifts unexpectedly during its lifetime (milliseconds)",
+		},
+		[]string{"platform", "user_agent"},
+	)
+	m.Registry.MustRegister(m.ClientCumulativeLayoutShift)
+
+	m.ClientLongTasks = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: MetricsNamespace,
+			Subsystem: MetricsSubsystemClients,
+			Name:      "long_tasks",
```

Comment:
The more I look at these metrics the more I wonder whether we should devise an additional bucket because most of these are likely to be web browser-specific and likely not applicable to other client types (i.e. mobile). I wonder if it would be valuable to make this distinction explicit. (e.g. adding a `web_` suffix or something).

Reply:
Maybe we should actually have a different subsystem for web vs mobile. While they're both clients, the metrics that they use will be fairly different, so it doesn't make sense to keep them together if we don't need to

Reply:
Yeah, not too bothered about the specifics but some stricter grouping would be helpful.

Reply:
Changed this to `MetricsSubsystemClientsWeb` and text to `webapp` to be more specific.